Added ColorPicker Component

Eshno opened the pull request:
## Description

Added ColorPicker Component

Issue https://github.com/yisselda/SlackableThemes/issues/12

**In case the component causes any crash into the app, please comment the steps to reproduce.
I'm accepting any suggestions on this, so feel free to comment!**

## Testing Steps: 
1. Clone my fork `git clone https://github.com/Eshno/SlackableThemes/tree/colorpicker`
2. Checkout into the `colorpicker` branch `git checkout colorpicker`
3. Run `npm i`
4. Head over into the `PageContainer.js` file
5. Import the **ColorPicker Component** `import ColorPicker from '../components/ColorPicker/';`
6. Add up the following code (image and snippet below)
7. _Verify_ that the component updates the color properly on any change
8. _Verify_ that the component closes when clicked outside of it


**Color Picker Component**

<img width="254" alt="Screen Shot 2019-10-06 at 10 09 24 AM" src="https://user-images.githubusercontent.com/34465493/66270436-9a315580-e821-11e9-933c-2720ede716b4.png">


**Testing Code (lines marked in green and blue)**

![Screen Shot 2019-10-06 at 9 57 09 AM](https://user-images.githubusercontent.com/34465493/66270339-3195a900-e820-11e9-914b-878029e3527d.png)

```
import React, {useState} from 'react'
import SideBar from '../components/SideBar';
import DashBoard from '../components/DashBoard';
import ColorPicker from '../components/ColorPicker/';

const PageContainer = props => {
  let [color, setColor] = useState('#00ffff');

  function updateColor(newColor) {
    setColor(newColor)
  }
  
  return (
    <div>
      <SideBar/>
      <DashBoard/>
      <span>{color}</span>
      <ColorPicker label="Color Picker" color={color} onChange={updateColor}/>
    </div>
    )
}
```

Eshno commented:
Please Squash and Merge this PR when its ready to merge!

yisselda commented:
@Eshno Thanks, I started trying out this component it looks good!
Can you add tests?

Eshno commented:
Sure, I'll be working on that soon

yisselda commented on src/tests/ColorPicker.test.js:
```diff
@@ -1,16 +1,16 @@
 import React from 'react';
 import Enzyme from 'enzyme';
 import { shallow } from 'enzyme';
-import ColorPickers from '../components/ColorPickers';
```
You need this line to import the component you are testing :)

Eshno replied:
Thank you! I'm just currently trying to make the test work properly, having some issues with it 🙂

Eshno commented:
@yisselda Did some small changes and added some tests on it, if anything misbehaves please tell me!

**Summary**:
- Added Tests
- Removed max length from the input (This was unnecessary)
- Small change on the Hex Validator (only verified the last digit in the string, now verifies the whole string)
